### ISSUE TITLE
no-unused-vars skips underscore arguments even when argsIgnorePattern does not match

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1227]
## Summary

Reproduced with @utoo/lint 0.4.1 on Node.js 22.23.1, macOS arm64. Neither _first nor _second matches /^NEVER$/. Both unused arguments should be reported. This also reproduces when argsIgnorePattern is omitted; there is no explicit /^_/ ignore setting.

## Minimal reproduction

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.1 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`example.tsx`:

```tsx
export function example(_first,_second) { return null; }
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "args": "after-used",
        "argsIgnorePattern": "^NEVER$"
      }
    ]
  }
}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "args": "after-used",
        "argsIgnorePattern": "^NEVER$"
      }
    ]
  }
};
```

```sh
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
```

## Result

ESLint reports two @typescript-eslint/no-unused-vars errors. utlint reports no diagnostics.

Both tools check the same file successfully; there are no syntax errors or lint-disable comments.
